src/js/components/dashboard/filters/DashboardAgencyFilter.jsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * DashboardAgencyFilter.jsx
+ * Created by Alisa Burdeyny 11/8/19
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Typeahead from 'components/SharedComponents/Typeahead';
+import ShownValue from './ShownValue';
+
+const propTypes = {
+    selectedFilters: PropTypes.object.isRequired,
+    onSelect: PropTypes.func.isRequired,
+    results: PropTypes.array.isRequired,
+    singleAgency: PropTypes.bool
+};
+
+const defaultProps = {
+    singleAgency: true
+};
+
+export default class DashboardAgencyFilter extends React.Component {
+    dataFormatter(item) {
+        return {
+            label: item.agency_name,
+            value: item.cgac_code ? item.cgac_code : item.frec_code
+        };
+    }
+
+    render() {
+        let selectedAgency = null;
+        let filteredList = this.props.results;
+        if (this.props.selectedFilters.agency !== '') {
+            const agencyCode = this.props.selectedFilters.agency;
+            // select only the agency we've selected for clearing and displaying
+            const agency = this.props.results.filter((result) => {
+                const code = result.cgac_code || result.frec_code;
+                return code === agencyCode;
+            });
+
+            // make the rest of the items so we don't have the chosen agency in the results list
+            filteredList = filteredList.filter((result) => {
+                const code = result.cgac_code || result.frec_code;
+                return code !== agencyCode;
+            });
+            // if it's a single agency user, we don't want to remove the value
+            const removeVal = this.props.singleAgency ? null : this.props.onSelect.bind(null, agencyCode);
```

Comment:
Recommend moving the binding out of `render`, since this creates a new function each time the component renders. Instead we can create `this.onSelect` func that references `this.props.selectedFilters.agency` and bind it in a constructor